src/dune_rules/odoc.ml

```diff
@@ -729,24 +717,12 @@ let gen_rules sctx ~dir:_ rest =
     match Package.Name.Map.find packages pkg with
     | None -> Memo.Build.return ()
     | Some _ -> setup_package_odoc_rules sctx ~pkg)
-  | "_odoc" :: "lib" :: lib :: _ -> (
-    let lib, lib_db = Scope_key.of_string sctx lib in
-    (* diml: why isn't [None] some kind of error here? *)
-    let* lib = Lib.DB.find lib_db lib in
-    match lib with
-    | None -> Memo.Build.return ()
-    | Some lib ->
-      (* TODO instead of this hack, call memoized function that generates the
-         rules for this library *)
-      let info = Lib.info lib in
-      let dir = Lib_info.src_dir info in
-      Load_rules.load_dir_and_produce_its_rules ~dir)
```

Comment:
Why is this dead code?  Wasn't it used for generating rules for private libraries?

Reply:
That's what I thought initially, but I added some debug printing code and this call to `load_dir_and_produce_its_rules` definitely doesn't produce any rule for this directory, so this code is definitely useless. (It's not dead-code, but it has no effect).

FTR, this the debug printing I did:

```ocaml
      let* rules =
        Rules.collect_unit (fun () ->
            Load_rules.load_dir_and_produce_its_rules ~dir)
      in
      Console.printf "private lib odoc rules for %s"
        (Lib.name lib |> Lib_name.to_string);
      Path.Build.Map.iteri (Rules.to_map rules) ~f:(fun p _ ->
          Console.printf "rules in %s" (Path.Build.to_string p));
      Rules.produce rules)
```

and then:

```
$ dune build _doc/_odoc/lib/dune_engine@160e75b64a67/x
private lib odoc rules for dune_engine
rules in _build/default
rules in _build/default/src/dune_engine
rules in _build/default/src/dune_engine/.dune_engine.objs
rules in _build/default/src/dune_engine/.dune_engine.objs/byte
rules in _build/default/src/dune_engine/.dune_engine.objs/native
rules in _build/default/src/dune_engine/.merlin-conf
Error: Don't know how to build _doc/_odoc/lib/dune_engine@160e75b64a67/x
```